junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,447 @@
+
+        def liftAnnotations(methodSymbol: Symbol): List[Tree] = {
+          val annotations = methodSymbol.annotations
+          def lift(t: Tree): Tree = {
+            t match {
+              case lit: Literal => lit
+
+              case Select(t2, name) => Select(lift(t2), name)
+
+              case ValDef(mod, name, tpt, rhs) =>
+                ValDef(mod, name, tpt, lift(rhs))
+
+              case _ => t
+            }
+          }
+          def liftList(params: List[Tree]): List[Tree] =
+            params.map(lift)
+
+          annotations.collect {
+            case ann if ann.atp == typeOf[org.junit.Test] =>
+              ann.original match {
+                case Block(stats, _) =>
+                  // TODO: Add support for @Test(timeout=...)
+                  // val newStats = stats
+                  // val newArgs = liftList(ann.args)
+                  // val newAnn = mkNewInstance[org.junit.Test](newArgs)
+                  // Block(newStats, newAnn)
+                  throw new UnsupportedOperationException(
+                    "@Test(timeout = ...) is not yet supported in " +
+                      "Scala.js JUnit Framework")
+
+                case _ => mkNewInstance[org.junit.Test](liftList(ann.args))
+              }
+
+            case ann if ann.atp == typeOf[org.junit.Before] =>
+              mkNewInstance[org.junit.Before]()
+
+            case ann if ann.atp == typeOf[org.junit.After] =>
+              mkNewInstance[org.junit.After]()
+
+            case ann if ann.atp == typeOf[org.junit.BeforeClass] =>
+              mkNewInstance[org.junit.BeforeClass]()
+
+            case ann if ann.atp == typeOf[org.junit.AfterClass] =>
+              mkNewInstance[org.junit.AfterClass]()
+
+            case ann if ann.atp == typeOf[org.junit.Ignore] =>
+              mkNewInstance[org.junit.Ignore](liftList(ann.args))
+
+            case ann if ann.atp == typeOf[org.junit.FixMethodOrder] =>
+              // TODO: Add support for @FixMethodOrder(...)
+              // mkNewInstance[org.junit.FixMethodOrder](liftList(ann.args))
+              throw new UnsupportedOperationException(
+                "@FixMethodOrder(...) is not yet supported in " +
+                  "Scala.js JUnit Framework")
+
+          }
+        }
+
+        def defaultMethodMetadata[T: TypeTag](methodAndId: (DefDef, Int)): Tree = {
+          val (m, id) = methodAndId
+          val annotations = liftAnnotations(m.symbol)
+          Apply(
+            Select(New(TypeTree(typeOf[T])), termNames_CONSTRUCTOR),
+            List(
+              Literal(Constant(m.name.toString)),
+              Literal(Constant(id.toString)),
+              mkList[java.lang.annotation.Annotation](annotations)
+            ))
+        }
+
+        def mkList[T: TypeTag](elems: List[Tree]): Tree = {
+          val array = ArrayValue(TypeTree(typeOf[T]), elems)
+          val castArray = gen.mkCast(array, typeOf[Array[Object]])
```

Comment:
There is no precomputed `Array[Object]` type, this seems to be as good as it can get.